tools/autograd/derivatives.yaml

```diff
@@ -1031,6 +1031,72 @@
 - name: special_zeta.other_scalar(Tensor self, Scalar other) -> Tensor
   self: not_implemented("zeta")
 
+- name: special_betaln(Tensor a, Tensor b) -> Tensor
+  a: grad * (at::digamma(a) - at::digamma(a + b))
+  b: grad * (at::digamma(b) - at::digamma(a + b))
+
+- name: special_betaln.a_scalar(Scalar a, Tensor b) -> Tensor
+  b: grad * (at::digamma(b) - at::digamma(a + b))
+
+- name: special_betaln.b_scalar(Tensor a, Scalar b) -> Tensor
+  a: grad * (at::digamma(a) - at::digamma(a + b))
+
+- name: special_betainc(Tensor self, Tensor a, Tensor b) -> Tensor
+  a, b, self: special_betainc_backward(grad, a, b, self)
+  result: "[&]() { auto [g_a, g_b, g_self] = betainc_partials(a_p, b_p, self_p);
+                   return g_a * a_t + g_b * b_t + g_self * self_t; }()"
+
+- name: special_betainc.x_scalar(Scalar self, Tensor a, Tensor b) -> Tensor
+  a, b: "[&]() { auto [g_a, g_b, g_self] = special_betainc_backward(grad, a, b, at::scalar_to_tensor(self, a.device()));
+                 return std::make_tuple(std::move(g_a), std::move(g_b)); }()"
+  result: "[&]() { auto [g_a, g_b, g_self] = betainc_partials(a_p, b_p, at::scalar_to_tensor(self, a_p.device()));
+                   return g_a * a_t + g_b * b_t; }()"
```

Comment:
I replaced a lot of verbose **backward** functions with a tensor tensor function receiving wrapped scalar values.

I don't remove the scalar overloads in fwd yet, because It's useful to provide c++ forward functions with scalar overloads.

And I'm not sure of where the code wrapping python number is located. (`torch/special/__init__.py` or `torch/_refs/special/__init__.py`?)
I don't think that `torch/_refs/special/__init__.py` is appropriate, because betainc, betaln and betaincinv cannot be decomposed. Then the other option is `torch/special/__init__.py`. I'm worried about verbose codes in `torch/special/__init__.py` which is a very simple file.

I think it might be okay to leave the scalar overloads for future c++ usage. What about you? @albanD

Reply:
> And I'm not sure of where the code wrapping python number is located. 

It is done by our code-generated python bindings. So when you call torch.foo(2.) in python, the python binding will nicely wrap that into a Tensor and at::foo will receive a Tensor.

Let me take another look at the code for the scalar overloads. But my current thinking is:
- C++ usage is relatively limited so I think it's fair to not provide everything there to keep things simpler.
- In particular, new entries here lead to quite a bit of codegen, python bindings and compilation time.
- If we do want to allow these, I think it is fair to add these conversions in the torch/include/* bindings for the C++ API by hand. This will reduce the amount of code/ops that we add while keeping the feature if users need it.